addons/event_crm/models/crm_lead.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api, _
+
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    event_id = fields.Many2one('event.event', string="Event linked to the lead")
+
+    registration_ids = fields.One2many('event.registration', 'lead_id', string="Attendees linked to the lead")
+    registration_count = fields.Integer(compute='_compute_registration_count', string="Number of registration linked to this lead")
+
+    @api.depends('registration_ids')
+    def _compute_registration_count(self):
+        for record in self:
+            record.registration_count = len(record.registration_ids)
+
+    def action_view_lead_registrations(self):
+        """
+        Allows to access at all the registrations/attendees linked to a lead
+        """
+        self.ensure_one()
+        action = self.env.ref('event.event_registration_action_tree').read()[0]
```

Comment:
Same as other action method: as you define a lot of things, sometimes it is better to define it in XML :) .